syntax = "proto3";

import "google/protobuf/any.proto";
//import "include/network/basic_type.proto";

package tcp_kit;

message BasicType {
    oneof value {
        uint32 u32 = 1;
        int32  s32 = 2;
        uint64 u64 = 3;
        int64  s64 = 4;
        float  f   = 5;
        double d   = 6;
        bool   b   = 7;
        string str = 8;
    }
}

message GenericMsg {
    string api = 1;
    repeated BasicType params = 2;
    optional google.protobuf.Any body = 3;
}